pymilvus/client/grpc_handler.py

```diff
@@ -653,6 +654,19 @@ def search_with_expression(self, collection_name, data, anns_field, param, limit
                                                      partition_names, output_fields, **_kwargs)
         _kwargs.pop("schema")
         _kwargs["auto_id"] = auto_id
+        # tag edit
+        round_decimal = param.pop("round_decimal",3)
+        if not isinstance(round_decimal,(int,str)):
+            raise ParamError("round_decimal must be int or str")
+        try:
+            round_decimal = int(round_decimal)
+        except Exception:
+            raise ParamError("round_decimal is not illegal")
```

Comment:
Should throw a BaseExecption exception.

Reply:
i see that other except also throws Exception ,for example in prepare.py 438line `try topK = int(topK) except Exception: raise ParamError("topk is not illegal") from None`. So i don't understand your meaning, repalce Exception with BaseException. If this , i will do. Thanks!

Reply:
Alright, it is also acceptable to use ParamError.